### ISSUE TITLE
[Table / Table Row] Ensure custom controls provide proper textual name, role, and state information (CAL-AMP54)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#13543]
**monday.com sync:** #10821950493

### Violation: 
 Ensure custom controls provide proper textual name, role, and state information 

### Recommended solution
Each table row should announce what record is being selected/unselected. We could provide a `label` to the `table-row` component that could provide this context.

<img width="1700" height="284" alt="Image" src="https://github.com/user-attachments/assets/1ab3b41b-9408-4158-9a06-a2f32bf11bcb" />

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG22/Understanding/non-text-content.html) 

### Severity: 
 High 

### Media Type: 
 Custom Controls 

--- 
### Areas for Remediation: 
- Title: CAL-AMP54 Ensure custom controls provide proper textual name, role, and state information 
- Module: M7 Table and color picker 
- Finding ID: [CAL-AMP54](https://esri.hub.essentia11y.com/short-link/LabETN_Ku18lG7f-) 

**Issue**
There are checkboxes (present beside the "Hiking trail" column) without appropriate names, roles, and state information. 

**User Impact**
Screen reader users will be unable to determine that these controls can be checked or unchecked. They will also be unable to review which options they have chosen.

**Code Reference** 
 ```
<calcite-table-header calcite-hydrated="" alignment="center"><th aria-colindex="2" class="selection-cell cell--multiple-selection selected-cell" role="columnheader" scope="col" tabindex="0">(...)</th></calcite-table-header>

```

**Steps to Reproduce**
1. Locate the checkbox.
2. Inspect it with Chrome DevTools.
3. In the Accessibility tab, expand the Computed Properties section.
4. Review the values for "Name", "Role", and "Checked".
 
**Recommendation**
Ensure custom controls provide proper textual name, role, and state information.

The best way to do this is to use native controls, as they come with this information built in. In this case, ```<input type="checkbox">``` elements would be most appropriate.

Developers can often achieve the same visual appearance using progressively enhanced CSS. Alternatively, they can visually hide the checkboxes with opacity: 0 and use SVGs to fake their appearance.

If conversion to native checkboxes is not possible, the following information is expected:
- The element functioning as the checkbox must have role="checkbox".
- This checkbox element must have a label provided by internal text, aria-label, or aria-labelledby.
- When checked, the checkbox element must have aria-checked="true", and when not checked, it must have aria-checked="false". If partially checked, it must have aria-checked="mixed".
- Checkboxes in a logical group must be children of an element with role="group". That element must have aria-labelledby set to the ID of text labelling the group.

**Recommended Reading**
For more information about creating accessible custom checkboxes, please see the W3C Web Accessibility Initiative's ARIA Authoring Practices Guide: https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/

**Compliant Code Example**
```
<!-- interactive anchor element that is in the focus order and has an href attribute-->
<a href="contact.html">Contact Us</a>
---
<!-- span element in the focus order with tabindex, keyboard pattern, and role -->
<span tabindex="0" role="link"> Contact Us</span>
```

 **Issue Image** 
 https://desktopqa/aic_lap/Calcite/Calcite2025/CAL-AMP54.png 
 

 --- 
### Additional Resources: 
- [LA Platform Report Link](https://esri.hub.essentia11y.com/short-link/LabETN_Ku18lG7f-) 
- [Esri Accessibility Resources](https://esriis.sharepoint.com/sites/Accessibility)
- Report Source: Product accessibility evaluation conducted on Calcite by Level Access and delivered August, 2025.
